cli/emit.rs

```diff
@@ -101,6 +101,28 @@ impl Emitter {
     }
   }
 
+  /// Expects a file URL, panics otherwise.
+  pub async fn load_and_emit_for_hmr(
+    &self,
+    specifier: &ModuleSpecifier,
+  ) -> Result<String, AnyError> {
+    let media_type = MediaType::from_specifier(specifier);
+    let source_code = tokio::fs::read_to_string(
+      ModuleSpecifier::to_file_path(specifier).unwrap(),
+    )
+    .await?;
+    let source_arc: Arc<str> = Arc::from(source_code.as_str());
```

Comment:
I don't think it clones under the hood, `Arc::from(source_code.into())` doesn't work here.

Reply:
I mean, try doing this:

```
let source_arc: Arc<str> = source_code.into()
```

I think maybe `.as_str()` does clone because I can use `source_code` after that line?